lib/Common/Memory/Recycler.h

```diff
@@ -758,7 +764,27 @@ class Recycler
     uint collectionFinishReason;
 #endif
 
-    CollectionState collectionState;
+    ObservableCollectionState collectionState;
+
+    inline void SetCollectionState(CollectionState newState)
+    {
+        this->collectionState = newState;
+    }
+
+    virtual void CollectionStateChanged(CollectionState newState, CollectionState oldState)
+    {
+#ifdef ENABLE_BASIC_TELEMETRY
+        if (this->collectionState == CollectionState::CollectionStateNotCollecting && newState != CollectionState::CollectionStateNotCollecting && newState != CollectionState::Collection_PreCollection)
```

Comment:
This code got me thinking about how are we going test this code to prevent regressions

Reply:
Was thinking that presence of telemetry data & its accuracy would indicate correctness.  Open to other suggestions.

Reply:
Yeah, but that will be a (very) lagging indicator?

---
In reply to: [187642759](https://github.com/Microsoft/ChakraCore/pull/5119#discussion_r187642759) [](ancestors = 187642759)

Reply:
one option is to dynamically add a ETW listener & confirm we see expected events for specific tests, but that feels overly-complex & introduces some dependencies into test frameworks that I'm less than excited about.  

My ultimate goal here is we're watching the telemetry signal from active branches + releases.  So if we have KPIs driven by this data, and if collection starts failing on our active dev branch , we'll know relatively quickly (e.g., within a couple of days).  This is predicated on some "team culture" assumptions, and I don't know how these will shake out, but it's my hope.

Happy to open an issue to track figuring out a testing strategy here.